### DISCUSSION TITLE
Can I keep cookies and site data without saving history?

### DISCUSSION THREAD

**Discussion by crocket** [qutebrowser/qutebrowser#6695] [Q&A] [answered]
I don't keep history. But, I still keep cookies and site data for some websites in firefox.

I wonder whether I can keep cookies and site data (for some websites) and discard history in qutebrowser, too.

`content.private_browsing` wipes cookies and site data, too.

**Comment by The-Compiler** [comment]
Can you elaborate on why? You can set `completion.web_history.max_items` if you just want the history to not be displayed.

**Reply by crocket** [reply]
cookies and site data are necessary for some websites. But, I don't want to keep cookies and site data for other websites.

I don't use history, and malwares can steal history to analyze me further if there is history record on disk.
I also don't store history for python REPL and bash after realizing that bash history contains sensitive data.

I wish qutebrowser could control storage of history, cookies, and site data separately.

If I want to revisit a web page later, I record it in my to-do task management system.

**Reply by crocket** [reply]
Here is an interesting read on web browser in general. It also talks about browser history.

https://lukesmith.xyz/articles/every-web-browser-absolutely-sucks

**Reply by The-Compiler** [reply]
It's deliberate that such an option doesn't exist, as it suggests a false sense of security. If you store cookies and site data on disk, someone can rather trivially reconstruct your browser history based on that, so disabling history storage won't really change much.

**Reply by crocket** [reply]
Site data would contain bank certificates which I need for sending money. Thus, it is necessary.
Searx also saves search preferences in cookies or site data. So, I want to keep cookies and site data for searx.

Apart from a few websites, I don't keep cookies and site data. I also don't store history.
Cookies and site data for only a few websites only give crude information to adversaries. History contains a lot more information than cookies and site data. For example, facebook may store login details as cookies, but facebook browsing history contains a lot more information than login details.

It is about privacy rather than security.

The key is to limit exposure as much as possible. I only keep what is necessary. A bit more fine-grained control over data storage can go a long way to limit exposure.

An option to wipe history at some point or not store history is going to be convenient. Is it already possible to keep cookies and site data only for some websites?

**Reply by crocket** [reply]
I came up with ideas.

* A job scheduler can be configured to execute commands when qutebrowser exits
  * `:history-clear` or `:cookie-delete except [example.com, domain2.com]` when qutebrowser exits.
* An option to keep history only in memory or disable history entirely without touching cookies and site data.

**Comment by crocket** [comment, marked as answer]
After reading qutebrowser subreddit, I realized qtwebengine saves its own history apart from qutebrowser and qutebrowser can't yet disable webengine history without private browsing mode.

I discovered that searx can encode its search preferences in URL parameters. I don't need cookies for searx.

Browsing history is obsolete. Allowing cookies and site data for a few select domains is desirable, but I think I will need to learn how to make things work without cookies and site data for now.
